packages/account-tree-controller/src/AccountTreeController.test.ts

```diff
@@ -1581,19 +1594,28 @@ describe('AccountTreeController', () => {
     });
 
     it('returns empty string when no wallets exist and getSelectedAccount returns EMPTY_ACCOUNT', () => {
-      const { controller, messenger } = setup({
+      const {
+        controller,
+        messenger,
+        accountsControllerMessenger,
+        accountTreeControllerMessenger,
+      } = setup({
         accounts: [],
         keyrings: [],
       });
 
       // Mock getSelectedAccount to return EMPTY_ACCOUNT_MOCK (id is '') BEFORE init
-      messenger.unregisterActionHandler(
+      accountsControllerMessenger.unregisterActionHandler(
         'AccountsController:getSelectedAccount',
       );
-      messenger.registerActionHandler(
+      accountsControllerMessenger.registerActionHandler(
         'AccountsController:getSelectedAccount',
         () => EMPTY_ACCOUNT_MOCK,
       );
+      messenger.delegate({
+        messenger: accountTreeControllerMessenger,
+        actions: ['AccountsController:getSelectedAccount'],
+      });
```

Comment:
If an action handler is unregistered and then re-registered, the root messenger must delegate the external action to the AccountTreeController messenger again, or the following error will occur: `A handler for AccountsController:getSelectedAccount has not been registered`